### PR TITLE
fix(MessageEmbed): include `author.name` in length getter

### LINKED ISSUE #5166: embed.length(): Missing author.name.length from character count
Hi! I was investigating how was `Embed.length()` calculated, for personal use.

From [Discord's docs](https://discord.com/developers/docs/resources/channel#embed-limits):
> Additionally, the characters in all `title`, `description`, `field.name`, `field.value`, `footer.text`, **and `author.name`** fields must not exceed 6000 characters in total. 

And in the [source code](https://github.com/discordjs/discord.js/blob/b0077dcd0ca53c28069aa60723e0f7e5f02f9b03/src/structures/MessageEmbed.js#L238-L247), I saw `Author.name`'s length isn't added to total number. Note I just *saw* it, didn't *test* if that can bring trouble (I don't use discord.js yet)

Goodbye!

### PR DESCRIPTION
**Pull request by vBread**
**Please describe the changes this PR makes and why it should be merged:**
Fixes #5166

**Status**

- [ ] Code changes have been tested against the Discord API, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**

- [ ] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/structures/MessageEmbed.js [hunk 1]
```diff
@@ -242,7 +242,8 @@ class MessageEmbed {
       (this.fields.length >= 1
         ? this.fields.reduce((prev, curr) => prev + curr.name.length + curr.value.length, 0)
         : 0) +
-      (this.footer ? this.footer.text.length : 0)
+      (this.footer ? this.footer.text.length : 0) +
+      (this.author ? this.author.name.length : 0)
```
Comment (papaia) [inline src/structures/MessageEmbed.js:246]:
```suggestion
      (this.footer?.text.length ?? 0) +
      (this.author?.name.length ?? 0)
```

File: src/structures/MessageEmbed.js [hunk 2]
```diff
@@ -231,7 +231,7 @@ class MessageEmbed {
   }
 
   /**
-   * The accumulated length for the embed title, description, fields and footer text
+   * The accumulated length for the embed title, description, fields, footer text and author name
```
Comment (papaia) [inline src/structures/MessageEmbed.js:234]: Oxford comma ¯\_(ツ)_/¯
```suggestion
   * The accumulated length for the embed title, description, fields, footer text, and author name
```

**Review by kyranet — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

File: src/structures/MessageEmbed.js [hunk 3]
```diff
@@ -242,7 +242,8 @@ class MessageEmbed {
       (this.fields.length >= 1
         ? this.fields.reduce((prev, curr) => prev + curr.name.length + curr.value.length, 0)
         : 0) +
-      (this.footer ? this.footer.text.length : 0)
+      (this.footer?.text.length ?? 0) +
+      (this.author?.name.length ?? 0)
```
Comment (NotSugden) [inline src/structures/MessageEmbed.js:246]: i think you should use the tenary operator here as before, makes more sense to me IMO (also this makes it inconsistent with how `this.title` and `this.description` length is retrieved here)

**Review by NotSugden — CHANGES_REQUESTED** [review]

File: src/structures/MessageEmbed.js (hunk 3 above)
Comment (iCrawl) [reply src/structures/MessageEmbed.js:246]: So the other 2 should be changed.

**Review by iCrawl — APPROVED** [review]

**Pull request merged**